src/commands/deployWorkspaceProject/DeployWorkspaceProjectSettings.ts

```diff
@@ -0,0 +1,37 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import type { WorkspaceFolder } from "vscode";
+import { settingUtils } from "../../utils/settingUtils";
+
+export interface DeployWorkspaceProjectSettings {
+    // Container app names are unique to a resource group
+    containerAppResourceGroupName?: string;
+    containerAppName?: string;
+
+    containerRegistryName?: string;
+}
+
+const deployWorkspaceProjectPrefix: string = 'deployWorkspaceProject';
+
+export async function getDeployWorkspaceProjectSettings(rootFolder: WorkspaceFolder): Promise<DeployWorkspaceProjectSettings | undefined> {
+    const settingsPath: string = settingUtils.getDefaultRootWorkspaceSettingsPath(rootFolder);
+
+    try {
+        const containerAppName: string | undefined = settingUtils.getWorkspaceSetting(`${deployWorkspaceProjectPrefix}.containerAppName`, settingsPath);
+        const containerAppResourceGroupName: string | undefined = settingUtils.getWorkspaceSetting(`${deployWorkspaceProjectPrefix}.containerAppResourceGroupName`, settingsPath);
+        const containerRegistryName: string | undefined = settingUtils.getWorkspaceSetting(`${deployWorkspaceProjectPrefix}.containerRegistryName`, settingsPath);
+
+        if (containerAppName || containerAppResourceGroupName || containerRegistryName) {
```

Comment:
This isn't a blocker, but I think it's awkward to only return an object if one of the properties is defined. And otherwise we return undefined.

In the future, if you add more settings, then you need to add them to this if expression.

I think I'd prefer if you just always return an object, even if the object just contains undefined values.

Reply:
Yeah I agree on it feeling a little awkward; the settings section felt a little clunky to me but I wasn't sure how to get around it without just changing the log outputs we display.

I kept that `undefined` path so that I could easily use this statement:
![image](https://github.com/microsoft/vscode-azurecontainerapps/assets/40250218/97a0f244-6f5e-44ef-9cc7-f9e259498719)

Though I've gone back and forth on whether it's even worth expressing that level of output detail to the user or not.  If I keep the statement and return an empty object, I'd just be moving that verbose `if` check up one level.  Is there a better way to handle this/what do you think?

Reply:
Yeah I agree, I'll have to think a little more about it, but the reason I was returning `undefined` like that was because it made it a lot easier to just check `if(!settings)` later.  I think if I remove the statement in this file, I'd then have to add a similar verbose check up one level to implement that logic path.  I'm wondering if there's a more clever way we can omit the verbose check entirely that I haven't though of yet.